test/rules/blas.jl

```diff
@@ -20,6 +20,41 @@ using LinearAlgebra.BLAS: gemm
                 x̄_ad = dx(ȳ)
                 x̄_fd = j′vp(central_fdm(5, 1), f, ȳ, x)
                 @test x̄_ad ≈ x̄_fd rtol=1e-9 atol=1e-9
+                if size(x) != ()  # A and B
+                    @test dx isa Rule{<:Function,<:Function}
+                    x̄ = zeros(size(x)...)
+                    ChainRules.accumulate!(x̄, dx, ȳ)
+                    @test x̄ ≈ x̄_ad rtol=1e-9 atol=1e-9
+                else  # α
+                    @test dx isa Rule{<:Function,Nothing}
+                end
+            end
+        end
+    end
+    @testset "gemv" begin
+        rng = MersenneTwister(2)
+        for n in 3:5, m in 3:5, t in ('N', 'T')
+            α = randn(rng)
+            A = randn(rng, m, n)
+            x = randn(rng, t === 'N' ? n : m)
+            y, (dt, dα, dA, dx) = rrule(gemv, t, α, A, x)
+            @test y ≈ α * (t === 'N' ? A : A') * x
+            @test dt isa ChainRules.DNERule
+            for (f, z, dz) in [(z->gemv(t, z, A, x), α, dα),
+                               (z->gemv(t, α, z, x), A, dA),
+                               (z->gemv(t, α, A, z), x, dx)]
+                ȳ = randn(rng, size(y)...)
+                z̄_ad = dz(ȳ)
+                z̄_fd = j′vp(central_fdm(5, 1), f, ȳ, z)
+                @test z̄_ad ≈ z̄_fd atol=1e-9 rtol=1e-9
+                if size(z) != ()  # A and x
+                    @test dz isa Rule{<:Function,<:Function}
+                    z̄ = zeros(size(z)...)
```

Comment:
Zero-ing out the cache in which we accumulate is dangerous since it leaves us vulnerable to bugs in which the accumulation rule _only_ works correctly if the cache is full of zeros.

Reply:
I'm not sure what you mean. This is only zeroing out in the test. Should we be using like `ones` or something instead?

Reply:
Yeah, preferably `randn` to avoid any particular deterministic structure in the values in the cache. If I'm not misremembering, we've had bugs in the past in Nabla whereby an in-place sensitivity accidentally zero-d out the cache, and we failed to pick this up in previous versions of the test suite because we were doing the above and only testing with caches whose values started at zero anyway.